third_party/vectors/sync.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python3
+
+import argparse
+import sys
+import pathlib
+import tomllib
+import tempfile
+import subprocess
+import typing
+import shutil
+import filecmp
+
+from vectorslib import utils
+from vectorslib.utils import SyncError
+
+
+def fetch_sources(
+    tmpdir: pathlib.Path,
+    sources: dict,
+    skip_fetch: bool,
+    using_custom_tmpdir: bool = False,
+):
+    if skip_fetch:
+        utils.info("skipping fetch")
+        return
+
+    for source_name, source_info in sources.items():
+        stmpdir = tmpdir / source_name
+
+        if stmpdir.is_dir():
+            assert using_custom_tmpdir
+            utils.warning(
+                f"using existing, potentially stale upstream clone of {source_name} at {stmpdir}"
+            )
```

Comment:
The naming here doesn't feel right which left me a bit confused. It's not that this is a "temporary directory" path that you are planning to clone into, instead this is an existing directory that is either a clone of upstream, or a source snapshot. I feel like maybe there is a better name for this? I think temporary is a built overloaded, perhaps offline_sources_path? Or something to that affect? It describes better how it is being used IMO.